internal/domain/course.go

```diff
@@ -8,32 +8,61 @@ import (
 // Status ...
 type Status uint8
 
-// Publish ...
 const (
-	Publish Status = iota + 1
-	Draft
-	Archive
+	CourseInDraft   string = "Draft"
```

Comment:
Better to build type CourseStatus (or something like this) string for this, so arbitrary string can't be messed up with status. This brings to us a benefit: detect bug earlier. 
```suggestion
type Status string

CourseInDraft  Status = "Draft"
```

Reply:
Done